src/state/therapies/reducer.ts

```diff
@@ -25,6 +25,10 @@ export default produce((draft: ITherapiesState, action: IAction<any, any>) => {
     case CREATE_THERAPY_SUCCESS: {
       draft.createTherapy.status = "SUCCESS";
       draft.createTherapy.data = action.payload;
+      draft.therapiesByPatientId.data = [
+        ...draft.therapiesByPatientId.data!,
+        action.payload,
+      ];
```

Comment:
The ! operator could cause some uncontrolled effects. It should more practical here to handler directly the case when the data will be null.

Reply:
You are right. I will replace it by 
`
 if (draft.therapiesByPatientId.data)
        draft.therapiesByPatientId.data = [
          ...draft.therapiesByPatientId.data,
          action.payload,
        ];
`

Reply:
If the condition is false, it's OK if I will not pass anything to the draft.therapiesByPatientId.data? Won't this cause any unhandled behaviours?

Reply:
I added the if() and else,  thanks.